src/main/java/org/zalando/riptide/RestBase.java

```diff
@@ -77,4 +88,43 @@ public D execute(final HttpMethod method, final URI url, final HttpHeaders heade
 
     protected abstract <T> D execute(final HttpMethod method, final URI url, final HttpEntity<T> entity);
 
+    static <T> void writeRequestEntity(final HttpEntity<T> entity, final ClientHttpRequest request, List<HttpMessageConverter<?>> converters) throws IOException {
+        final HttpHeaders headers = entity.getHeaders();
+        request.getHeaders().putAll(headers);
+
+        @Nullable final T body = entity.getBody();
+
+        if (body != null) {
+
+            final Class<?> type = body.getClass();
+            @Nullable final MediaType contentType = headers.getContentType();
+
+            final Optional<HttpMessageConverter<T>> match = converters.stream()
```

Comment:
> I checked and Spring is only doing this when reading responses, but not for requests

Do you have a Link? I only found https://github.com/spring-projects/spring-framework/blob/master/spring-web/src/main/java/org/springframework/web/client/HttpMessageConverterExtractor.java#L87 which is just going through the list.

Only sorting I found was for accept headers in https://github.com/spring-projects/spring-framework/blob/master/spring-web/src/main/java/org/springframework/web/client/RestTemplate.java#L744